CoreGraphics.Drawing build fails on reference platform

Doesn't seem to be happy about unique_cf

CGContextDrawing_ImageMaskTests.cpp:167:36: note: in instantiation of function template specialization 'std::__1::unordered_map<unsigned int,
      woc::unique_cf<CGImage *>, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<std::__1::pair<const
      unsigned int, woc::unique_cf<CGImage *> > > >::emplace<int, woc::unique_cf<CGImage *> >' requested here
                    clippingImages.emplace(newShape | newType, std::move(clipImage));
                                   ^
..//../../include/xplat/Starboard/SmartTypes.h:541:19: note: copy constructor is implicitly deleted because 'unique_cf<CGImage *>' has a
      user-declared move assignment operator
    unique_cf<T>& operator=(unique_cf<T>&& other) = delete;
                  ^
1 error generated.
make: *** [..//../../build/Tests/UnitTests/CoreGraphics.Drawing/OSX/Objects/CGContextDrawing_ImageMaskTests.cpp.o] Error 1

--- a/include/xplat/Starboard/SmartTypes.h
+++ b/include/xplat/Starboard/SmartTypes.h
@@ -528,6 +528,9 @@ public:
     unique_cf(T val): AutoCF<T, CFLifetimeRetain>(std::move(val)) {
     }
 
+    unique_cf(unique_cf<T>&& other): AutoCF<T, CFLifetimeRetain>(std::move(other)) {
+    }
+
     void reset(T val = nullptr) {
         AutoCF<T, CFLifetimeRetain>::attach(val);
     }
@@ -538,7 +541,11 @@ public:
 
     unique_cf<T>& operator=(T val) = delete;
     unique_cf<T>& operator=(const unique_cf<T>& other) = delete;
-    unique_cf<T>& operator=(unique_cf<T>&& other) = delete;
+
+    unique_cf<T>& operator=(unique_cf<T>&& other) {
+        this->AutoCF<T>::operator=(std::move(other));
+        return *this;
+    }
 };
 
 template <typename T = CFTypeRef>

--- a/tests/UnitTests/Starboard/AutoCFTests.cpp
+++ b/tests/UnitTests/Starboard/AutoCFTests.cpp
@@ -157,3 +157,31 @@ TEST(AutoCF, LegacyUniqueCF) {
     // Only one retain, even though the default ctor is being used.
     ASSERT_EQ(1, CFGetRetainCount(testObject));
 }
+
+TEST(AutoCF, LegacyUniqueCFMove) {
+    woc::unique_cf<CFTestClassRef> testObject{ CFTestClassCreate(nullptr) };
+
+    // Only one retain, even though the default ctor is being used.
+    ASSERT_EQ(1, CFGetRetainCount(testObject));
+
+    woc::unique_cf<CFTestClassRef> testMovedObject{ std::move(testObject) };
+
+    // Still only one retain.
+    ASSERT_EQ(1, CFGetRetainCount(testMovedObject));
+}
+
+TEST(AutoCF, LegacyUniqueCFMoveAssignment) {
+    woc::unique_cf<CFTestClassRef> testMovedObject{};
+
+    {
+        woc::unique_cf<CFTestClassRef> testObject{ CFTestClassCreate(nullptr) };
+
+        // Only one retain, even though the default ctor is being used.
+        ASSERT_EQ(1, CFGetRetainCount(testObject));
+
+        testMovedObject = std::move(testObject);
+    }
+
+    // Still only one retain.
+    ASSERT_EQ(1, CFGetRetainCount(testMovedObject));
+}

